v3/src/components/graph/utilities/graph-utils.ts

```diff
@@ -1,18 +1,22 @@
 import {format} from "d3"
 import * as PIXI from "pixi.js"
 import {IDataSet} from "../../../models/data/data-set"
+import { ITileModel } from "../../../models/tiles/tile-model"
+import { toV2Id } from "../../../utilities/codap-utils"
 import {
   defaultSelectedColor, defaultSelectedStroke, defaultSelectedStrokeWidth, defaultStrokeWidth
 } from "../../../utilities/color-utils"
 import { isFiniteNumber } from "../../../utilities/math-utils"
 import { t } from "../../../utilities/translation/translate"
 import {IAxisModel, isBaseNumericAxisModel} from "../../axis/models/axis-model"
-import {ScaleNumericBaseType} from "../../axis/axis-types"
+import {AxisOrientation, ScaleNumericBaseType} from "../../axis/axis-types"
+import { GraphPlace } from "../../axis-graph-shared"
 import {CaseData} from "../../data-display/d3-types"
 import {Point, PointDisplayType, transitionDuration} from "../../data-display/data-display-types"
 import {IDataConfigurationModel} from "../../data-display/models/data-configuration-model"
 import {IPixiPointMetadata, PixiPoints} from "../../data-display/pixi/pixi-points"
 import { IGraphDataConfigurationModel } from "../models/graph-data-configuration-model"
+import { IGraphContentModel } from "../models/graph-content-model"
```

Comment:
This is causing a circular dependency warning. I'm leaving it as is till I'm more certain of the overall structure of things.

Reply:
Rather than putting `attrChangeNotificationValues` and friends in this module, I'd put them in a new `graph-notifications.ts` module, which should avoid the circular reference. This module is too low-level to contain a function that requires access to the `GraphContentModel`.